SSA/Experimental/Bits/Fast/README.md

```diff
@@ -0,0 +1,94 @@
+# Automata Tactic
+
+This is documentation for how the automata tactic works. This tactic deserves documentation because it is non-trivial and contains some difficult concepts.
+
+By analogy, the automata tactic is like Presberger Arithmetic. Just like Presberger Arithmetic only involves constant-memory operations on natural numbers,
+the Automata Tactic can automatically solve expressions on BitVectors which involve constant-memory operations.
+
+# Goal of the Automata Tactic
+
+The automata tactic is a decision procedure. It takes in statments of the form
+
+
+\[
+\forall (w : \mathbb{N})
+\forall (x_1, x_2, x_3, \ldots x_n : BitVec w)
+(x_1 &&& x_2) + (x_1 ||| x_2) = (x_1 + x_2)
+\]
+
+The automata tactic only accepts expressions containing finite memory operations, or operations which can 
+be implmeneted with an amount of extra memory scratch space which does not depend on the size of the inputs.
+For example, the automata tactic works with
+
+1) Bitwise operations (which require 0 extra bits of memory scratch space)
+2) Addition and subtraction (which require 1 extra bit of memory scratch space)
+3) negation (which requires 1 extra bit of memory scratch space)
+
+But crucially, the automata tactic does not support multiplication, because the amount of memory
```

Comment:
```suggestion
3) Negation (which requires 1 extra bit of memory scratch space)

But crucially, the automata tactic does not support multiplication, because the amount of memory
```